bson/empty_interface_codec.go

```diff
@@ -8,47 +8,24 @@ package bson
 
 import (
 	"reflect"
-
-	"go.mongodb.org/mongo-driver/bson/bsonoptions"
 )
 
-// EmptyInterfaceCodec is the Codec used for interface{} values.
-//
-// Deprecated: Use [go.mongodb.org/mongo-driver/bson.NewRegistry] to get a registry with the
-// EmptyInterfaceCodec registered.
-type EmptyInterfaceCodec struct {
-	// DecodeBinaryAsSlice causes DecodeValue to unmarshal BSON binary field values that are the
+// emptyInterfaceCodec is the Codec used for interface{} values.
+type emptyInterfaceCodec struct {
+	// decodeBinaryAsSlice causes DecodeValue to unmarshal BSON binary field values that are the
 	// "Generic" or "Old" BSON binary subtype as a Go byte slice instead of a Binary.
-	//
-	// Deprecated: Use bson.Decoder.BinaryAsSlice instead.
-	DecodeBinaryAsSlice bool
+	decodeBinaryAsSlice bool
 }
 
 var (
-	defaultEmptyInterfaceCodec = NewEmptyInterfaceCodec()
-
-	// Assert that defaultEmptyInterfaceCodec satisfies the typeDecoder interface, which allows it
+	// Assert that emptyInterfaceCodec satisfies the typeDecoder interface, which allows it
 	// to be used by collection type decoders (e.g. map, slice, etc) to set individual values in a
 	// collection.
-	_ typeDecoder = defaultEmptyInterfaceCodec
+	_ typeDecoder = (*emptyInterfaceCodec)(nil)
```

Comment:
Suggest inlining the variable and using the conventional type assertion pattern: `var _ interface = &struct{}`